src/hardware/driver/dac.rs

```diff
@@ -0,0 +1,240 @@
+///! Driver DAC11001 driver
+use core::fmt::Debug;
+use embedded_hal::blocking::spi::{Transfer, Write};
+use stm32h7xx_hal::gpio;
+
+use super::ChannelVariant;
+
+// DAC11001 register addresses
+#[allow(unused, non_camel_case_types)]
+pub enum DAC_ADDR {
+    NOP = 0x00 << 24,
+    DAC_DATA = 0x01 << 24,
+    CONFIG1 = 0x02 << 24,
+    DAC_CLEAR_DATA = 0x03 << 24,
+    TRIGGER = 0x04 << 24,
+    STATUS = 0x05 << 24,
+    CONFIG2 = 0x06 << 24,
+}
+
+#[allow(non_snake_case)]
+pub mod CONFIG1 {
+    pub mod EN_TMP_CAL {
+        pub const DISABLED: u32 = 0 << 23;
+        pub const ENABLED: u32 = 1 << 23;
+    }
+    pub mod TNH_MASK {
+        pub const JUMP14: u32 = 00 << 18;
+        pub const JUMP15: u32 = 01 << 18;
```

Comment:
There are a couple clippy warnings here.
Pretty sure many of your numbers are actually wrong. If you want binary, write `0b01`.